todos/src/App.jsx

```diff
@@ -1,10 +1,62 @@
-import "./App.css";
+import Todo from "./components/Todo";
+
+import { faCirclePlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useState } from "react";
+
+import { useDispatch, useSelector } from "react-redux";
+import { addTodo } from "./redux/todoSlice";
 
 function App() {
+  const dispatch = useDispatch();
+  const todos = useSelector((state) => state.todos);
+  const [newTodo, setNewTodo] = useState("");
+
+  const addNewTodo = () => {
+    if (newTodo.trim() !== "") {
+      dispatch(addTodo(newTodo));
+      setNewTodo("");
+    }
+  };
+
+  const endTask = (index) => {
+    dispatch(endTask(index));
+  };
+
+  const deleteTask = (id) => {
+    dispatch(deleteTask(id));
+  };
+
   return (
-    <>
-      <h1 className="bg-red-500">Hello there</h1>
-    </>
+    <div className="text-center">
+      <h1 className="py-10 text-5xl font-bold text-gray-200">Todos</h1>
+      <div className="flex items-center relative w-1/2 m-auto">
+        <input
+          className="border w-full rounded-full p-2 focus:border-none"
+          type="text"
+          placeholder="Add todo..."
+          value={newTodo}
+          onChange={(e) => setNewTodo(e.target.value)}
+        />
+        <FontAwesomeIcon
+          className="absolute right-2 cursor-pointer"
+          icon={faCirclePlus}
+          color="green"
+          onClick={addNewTodo}
+        />
+      </div>
+      {todos.map((todo, index) => (
+        <Todo
+          key={index}
+          id={todo.id}
+          index={index}
+          todo={todo.todo}
```

Comment:
Suggestion: having a property called `todo` inside a todo is somehow confusing because of that repetition, I would recommend something like `title` or `name` or something similar to explain the content of that todo
for example
```js
const todo = {
  title: "Cleaning the room",
  hasEnded: false
}
```
But this is optional  having `todo` inside is also not that bad.